src/components/shared/Slider/index.tsx

```diff
@@ -13,21 +16,46 @@ interface Props {
   step?: number;
 }
 
-const Slider: FC<Props> = () => {
+const Slider: FC<Props> = ({ maxValue = 100, minValue = 0 }) => {
+  const sliderRef = useRef<any>();
+  const [sliderWidth, setSliderWidth] = useState(0);
+  const [layoutX, setLayoutX] = useState(0);
+  const [positionX, setPositionX] = useState(0);
+
+  const getPercentagePosition = (): number => {
+    const percentage = (positionX / sliderWidth) * 100;
+    if (percentage <= 0) return 0;
+    else if (percentage >= 100) return 100;
+    else return percentage;
+  };
+
   const panResponder = React.useMemo(
     () =>
       PanResponder.create({
-        onStartShouldSetPanResponder: () => true,
-        onPanResponderGrant: (evt, gestureState) => {
-          console.log(gestureState);
+        onMoveShouldSetPanResponder: () => true,
+        onPanResponderMove: (evt, gestureState) => {
+          const moveX = gestureState.moveX;
+          setPositionX(moveX - layoutX);
         },
       }),
-    [],
+    [layoutX],
   );
 
   return (
-    <Container {...panResponder.panHandlers}>
+    <Container
+      ref={sliderRef}
+      {...panResponder.panHandlers}
+      onLayout={(): void => {
+        if (sliderRef) {
+          sliderRef.current.measure((x, y, width, height, pageX) => {
```

Comment:
I don't understand this code, cause of my lack of knowledge.
Could you please describe it simply?
`setLayoutX(pageX)`
`setSliderWidth(width)`

Reply:
I don't know about penResponder API either.  I tried to use `moveX` property to get the touched coordinations. But it's absolute value so I have to subtract slider's X coordination.  The layoutX is the slider's X coordination.
percentagePosition is the percentage of value.

Reply:
Thanks for your description!

From your description,
I have opinions for the names.


Slider is one dimension. That means not Y, and X even if it is vertical.
So, How about just `value` as you implemented the [Slider] interface https://github.com/dooboolab/dooboo-ui-native/issues/47?

I think `sliderWidth` is better to use.
Width is a widely used design unit than length to fit the screen width.
What do you think about it?

`value` is same to `percent`.
`percent` value has to be parsed to use every time to use as a display value for user experience.
The prop `value` type of [Slider] is an integer.
So, How about using `value` as you implemented the [Slider] interface https://github.com/dooboolab/dooboo-ui-native/issues/47?

About `sliderPositionX`,
That makes coupling with the parent components even with the screen.
It makes hard to change and maintain later.
I think the position for a component should be defined in the parent component.

Good to refer about cohesion and coupling:
https://freefeast.info/difference-between/difference-between-cohesion-and-coupling-cohesion-vs-coupling/

Reply:
Thank you for kind description.

1.`sliderWidth`
Good point! I haven't think about it deeply as you. I searched what's the different between `width` and `length`.

>1.Length is describing how long something is while width is describing how wide an object is.
2.In geometry, length pertains to the longest side of the rectangle while width is the shorter side.
3.Length can also refer to an extent of time or a measure of distance.

Read more: Difference Between Length and Width | Difference Between http://www.differencebetween.net/science/mathematics-statistics/difference-between-length-and-width/#ixzz68E5IrKvS

So I think for screen `width` is good but in this case, probably `length` is more appropriate. What do you think about it?

2.`value`
I am not sure I understand you exactly. 
For the slider interface, I use prop like `default value`. But anyway to draw Track, Thumb, even Label we should know `percent` value to set`left` CSS property value.

3.`sliderPositionX `
Yes. I agree with your opinion that the position for a component should be defined in the parent component. But I think it doesn't have a dependency on the parent component. `sliderPositionX` is set after rendering by `measure` function. I don't get `sliderPositionX ` as props.

Reply:
1. I make sense of your description and the references. I think they are a general description not about our work. UI components have a relation in their positions and sizes. `width` is a unit to compare and fit a proper position among components like screen, slider or other UI components.
`length` may confuse other developers in comparing the sizes between a parent component and slider, because they have different units `length` and `width`.
In general for programming, `length` is used for count of array-like object. (array, string)

2. Oh, sorry for that I thought it is passed as a prop. And I see now that `percent` is internally used for the position as a component state. :)

3. Thank you for the description. :)
I got where `sliderPositionX` is from.

@YongpilMoon , I think we should have a discussion with @daheeahn on slack or here. And @daheeahn is implementing [Track] with the gesture behavior with the positioning system. I guess she is considering 2 and 3 features as [Track]'s roles because I described the features as `Track`'s roles in https://github.com/dooboolab/dooboo-ui-native/issues/41 from the start.

Reply:
1. Yes, it makes sense. Thanks for your good description. I will fix it.
> length is used for count of array-like object. (array, string)

Okey. Let's merge after @daheeahn check it.

Reply:
I think we can't wait until she checks it. And we need to decide the structure to progress more.
Why don't we just merge the pull requests and focus on complete fast? I think we can get a better idea and modify the structure during the process. @Sandwichj